### ISSUE TITLE
Suggestion: Notes and Credits/Instructions API

### ISSUE THREAD

**Issue by AmazingMech2418** [LLK/scratch-gui#2200]
### Expected Behavior

If the suggestion is accepted, in the editor, there would be a way to code the Notes and Credits and Instructions to change based on viewer preferences.

### Actual Behavior

There is no way to code the Notes and Credits or Instructions. For example, if you want to allow anybody to understand the instructions, you can't use the new Google Translate extension to translate them.

### Steps to Reproduce

It always happens.

### Operating System and Browser

It doesn't matter.

Here's my request in more detail:
In a similar way to the backdrop coding area, there should be Notes and Credits and Instructions coding areas. They would contain blocks like these:
![scratchblocks 4](https://user-images.githubusercontent.com/33498997/40806212-d12500ee-64ee-11e8-81c2-13bdc2f86aed.png)
Note: The image is from https://s3blocks.github.io/#set%20%5B%5D%20to%20%5BNotes%20and%20Credits%20v%5D%3A%3A%2330f030%0Acontent%20of%20%5BNotes%20and%20Credits%20v%5D%3A%3Areporter%20%2330f030%0AWhen%20Page%20Loads%3A%3A%2330f030%20hat%0A%0A%2F%2FThis%20is%20a%20concept%20idea%20of%20the%20Notes%20and%20Credits%20and%20Instructions%20section%20for%20Scratch%203.%20%0A%2F%2F%20It%20will%20also%20exclude%20broadcast%20blocks%2C%20so%20people%20don't%20make%20unvoluntary%20actions%20in%20projects...%0A%2F%2F%20...%20with%20the%20%22When%20Page%20Loads%22%20block.%20%0A%0A%2F%2F%20Option%20for%20translations%3A%0A%0AWhen%20Page%20Loads%3A%3A%2330f030%20hat%0Aset%20%20(translate%20(content%20of%20%5BNotes%20and%20Credits%20v%5D%3A%3Areporter%20%2330f030)%20to%20(viewer%20language%3A%3Areporter))%20to%20%5BNotes%20and%20Credits%20v%5D%3A%3A%2330f030%0Aset%20%20(translate%20(content%20of%20%5BInstructions%20v%5D%3A%3Areporter%20%2330f030)%20to%20(viewer%20language%3A%3Areporter))%20to%20%5BInstructions%20v%5D%3A%3A%2330f030%0A

**Comment by CosmicWebServices** [comment]
Interesting concept but there are a bunch of problems I see already. Adding on to @joker314 allowing projects to run when the page loads can be a bad idea... But then if that is out of the question it makes the notes and credits block useless. Another problem is that you are going to end up with really long blocks because people will have to type in the whole text in the notes and credits. Also, imagine cloud blocks + these blocks...bad idea practically a chat box.

**Comment by amazinigmech2418** [comment]
@joker314 Only those blocks, hat blocks, and reporters would be allowed. No stacks for variables and messages will be added.

**Comment by amazinigmech2418** [comment]
@CosmicWebServices See the comment to @joker314 . Also, cloud doesn't support text. It is mainly for translation, so you don't have to go to google translate and type in the whole notes and credits. It can also be good for personalization in game instructions.

**Comment by joker314** [comment]
It's an interesting idea, but:
- What kind of error should we show (or not) to indicate that you tried to run a not-allowed block? 
- So presumably you're saying the "translate" block is allowed. If it's the only allowed block, then why not add a built-in translation feature in the Instructions/Notes and Credits?

I guess my concern is that Google Translate is inaccurate. Some users may want to either improve foreign language skills; or, more likely, want to see the original to actually understand what's going on.

**Comment by amazinigmech2418** [comment]
@joker314 It wouldn't allow you to drop a non-allowed block in the coding area. There should be all extensions built in to Notes and Credits and Instructions so the coder doesn't have to use extensions.

**Comment by joker314** [comment]
The user should have a choice to see the original Notes and Credits and Instructions, and it's massively unfair to disallow the user from doing that. I'm not sure what you're saying, you mean the translation extension should be duplicated into these blocks?

I'm okay with a little button at the bottom of the Notes and Credits that says "Translate with Google Translate" or something like that; but we shouldn't be doing this forcefully.

**Comment by amazinigmech2418** [comment]
@joker314 It could also allow someone to use their username in the notes and credits/instructions to personalize the instructions more. Also, what if someone wants to make instructions that adapt to where you are in a game?

**Comment by kyleplo** [comment]
I don't think this is a good idea, but maybe a way to personalize notes/credits and instructions. For example, "Hi, _user_" in notes/credits or instructions would become "Hi, @kyleplo"

**Comment by amazinigmech2418** [comment]
@kyleplo Why don't you think it is a good idea? It could allow for translation, personalization, and adaptations to the instructions and notes and credits.

**Comment by kyleplo** [comment]
@amazinigmech2418 Well, it's not entirely bad, but I don't see the Scratch team ever approving this.

**Comment by amazinigmech2418** [comment]
@kyleplo Why?
